src/main/java/seedu/address/logic/commands/EditApplicantCommand.java

```diff
@@ -0,0 +1,151 @@
+package seedu.address.logic.commands;
+
+import static java.lang.Integer.parseInt;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICANT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_APPLICANT_WITH_ID;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.listing.Listing;
+
+/**
+ * Edits a listing in the listing book.
+ */
+public class EditApplicantCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit_app";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits an applicant's details for a listing. "
+            + "by the index number of the listing used in the displayed listing book. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_APPLICANT_WITH_ID + "OLD_APPLICANT]...\n"
+            + "[" + PREFIX_APPLICANT + "NEW_APPLICANT]...\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_APPLICANT_WITH_ID + "John "
+            + PREFIX_APPLICANT + "Sam";
+
+    public static final String MESSAGE_SUCCESS = "Applicant: %1$s has been edited to %2$s in %3$s";
+    public static final String MESSAGE_APPLICANT_NOT_FOUND = "Applicant %1$s cannot be found in %2$s.";
+    public static final String MESSAGE_DUPLICATE_APPLICANT = "This listing already contains the provided applicant";
+    public static final String MESSAGE_AMBIGUOUS_APPLICANT = "There are multiple applicants with the name %1s in %2$s, "
+            + "specify the 4-digit "
+            + "unique identifier after the name.\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_APPLICANT_WITH_ID + "John Doe#2103 "
+            + PREFIX_APPLICANT + "Sam";
+    private final Index targetIndex;
+    private final String oldApplicantId;
+
+    private final Applicant newApplicant;
+
+    /**
+     * Creates an EditCommand to edit the specified {@code Listing}
+     */
+    public EditApplicantCommand(Index targetIndex, String oldApplicantId, Applicant newApplicant) {
+        requireNonNull(targetIndex);
+        requireNonNull(oldApplicantId);
+        requireNonNull(newApplicant);
+
+        this.targetIndex = targetIndex;
+        this.oldApplicantId = oldApplicantId;
```

Comment:
if the user cannot see the id how ah, is that case handled?

Reply:
they will be able to see the id if there is a need for it.
![image](https://user-images.githubusercontent.com/85011179/227136403-36a051c9-dd99-4e7c-8d9f-9192f622d6fa.png)